src/Type/Php/ArrayChunkFunctionReturnTypeExtension.php

```diff
@@ -55,10 +58,17 @@ public function getTypeFromFunctionCall(FunctionReflection $functionReflection,
 		}
 
 		$constantArrays = $arrayType->getConstantArrays();
-		if ($lengthType instanceof ConstantIntegerType && $lengthType->getValue() >= 1 && $preserveKeys !== null && count($constantArrays) > 0) {
+		$biggerOne = IntegerRangeType::fromInterval(1, null);
+		if (count($constantArrays) > 0 && $biggerOne->isSuperTypeOf($lengthType)->yes()) {
 			$results = [];
 			foreach ($constantArrays as $constantArray) {
-				$results[] = $constantArray->chunk($lengthType->getValue(), $preserveKeys);
+				foreach ($lengthType->getFiniteTypes() as $finiteType) {
+					if (!$finiteType instanceof ConstantIntegerType) {
+						throw new ShouldNotHappenException();
+					}
+
+					$results[] = $constantArray->chunk($finiteType->getValue(), $preserveKeys);
```

Comment:
I guess this might get slow when big integer ranges are involved for `$length`. any opinions about possible limits or different implementations?